src/kubernetes.cc

```diff
@@ -982,20 +1004,22 @@ void KubernetesUpdater::StartUpdater() {
     auto cb = [=](std::vector<MetadataUpdater::ResourceMetadata>&& results) {
       MetadataCallback(std::move(results));
     };
-    node_watch_thread_ = std::thread([=]() {
+    object_watch_threads_.emplace(WatchId("nodes", "v1"), std::thread([=]() {
       reader_.WatchNodes(watched_node, cb);
```

Comment:
Just realized that it's probably more efficient to be capturing `reader_` by reference everywhere. But we'll do that as a separate update, as it's orthogonal to this PR.

Reply:
ack.